src/addon/validation.cpp

```diff
@@ -30,6 +30,16 @@ namespace {
 		"other", ""
 	};
 
+	// Reserved DOS device names on Windows XP and later.
+	const std::set<std::string> dos_device_names = {
+		"NUL", "CON", "AUX", "PRN",
+		// Console API devices
+		"CONIN$", "CONOUT$",
+		// Configuration-dependent devices
+		"COM1", "COM2", "COM3", "COM4", "COM5", "COM6", "COM7", "COM8", "COM9",
+		"LPT1", "LPT2", "LPT3", "LPT4", "LPT5", "LPT6", "LPT7", "LPT8", "LPT9",
```

Comment:
There are so many numbered device names now that instead of hardcoding them, I'd prefer checking for DOS names with `(stem.find("COM") == 0 || stem.find("LPT") == 0) && stem[3] >= '0' && stem[3] <= '9'`

Reply:
The list isn't going to change any time soon and it's easier to tell what the items are this way.